ignite/handlers/param_scheduler.py

```diff
@@ -16,6 +17,305 @@
 
 
 class ParamScheduler(metaclass=ABCMeta):
+    """An abstract class for updating a parameter value during
+    training.
+
+    Args:
+        param_name: name of parameter to update.
+        save_history: whether to log the parameter values to
+            `engine.state.param_history`, (default=False).
+
+
+    .. versionadded:: 0.5.1
+    .. versionchanged:: 0.6.0
+    """
+
+    def __init__(
+        self, param_name: str, save_history: bool = False,
+    ):
+        self.param_name = param_name
+        self.event_index = 0
+        self._save_history = save_history
+        self._state_attrs = ["event_index", "param_name", "save_history"]
+
+    @abstractmethod
+    def __call__(self, engine: Optional[Engine], name: Optional[str] = None) -> None:
+        pass
+
+    @property
+    def save_history(self) -> bool:
+        return self._save_history
+
+    @save_history.setter
+    def save_history(self, value: bool) -> None:
+        self._save_history = value
+
+    def state_dict(self) -> Dict[str, Any]:
+        """Returns a dictionary containing a whole state of ParamScheduler.
+
+        Returns:
+            dict:
+                a dictionary containing a whole state of ParamScheduler
+        """
+        destination = OrderedDict()
+        for name in self._state_attrs:
+            if hasattr(self, name):
+                val = getattr(self, name)
+                if hasattr(val, "state_dict"):
+                    val = val.state_dict()
+                destination[name] = copy(val)
+        return destination
+
+    def load_state_dict(self, state_dict: Mapping) -> None:
+        """Copies parameters from :attr:`state_dict` into this ParamScheduler.
+
+        Args:
+            state_dict: a dict containing parameters.
+        """
+        if not isinstance(state_dict, Mapping):
+            raise TypeError(f"Argument state_dict should be a dictionary, but given {type(state_dict)}")
+
+        for name in self._state_attrs:
+            if name not in state_dict:
+                raise ValueError(
+                    f"Required state attribute '{name}' is absent in provided state_dict '{state_dict.keys()}'"
+                )
+            val = state_dict[name]
+            obj = getattr(self, name)
+            if isinstance(val, Mapping) and hasattr(obj, "load_state_dict"):
+                obj.load_state_dict(val)
+            else:
+                setattr(self, name, val)
+
+    @abstractmethod
+    def get_param(self) -> Union[List[float], float]:
+        """Method to get current optimizer's parameter values
+
+        Returns:
+            list of params, or scalar param
+        """
+        pass
+
+
+class AnyParameterScheduler(ParamScheduler):
```

Comment:
Instead of `Any*` we can call them `State*` ?

Reply:
yes `State*` sounds good